Remove app module

Stentsoe opened the pull request:
The app module is not needed and can therefore be removed. The modem and ui module now initiates without the APP_EVT_START event.

Riphiphip reviewed (dismissed):
MSC files and ui module changes should probably be their own PR, but it should be fine for now.

Stentsoe committed:
Remove app module

The app module is not needed and can therefore be removed. The modem and ui module now initiates without the APP_EVT_START event.

Riphiphip reviewed (approved):
🏎️ 💨
